services/state/token/account.proto

```diff
@@ -0,0 +1,176 @@
+     */
+    bool smart_contract = 8;
+    /**
+     * A boolean marking if the account requires a receiver signature.
+     */
+    bool receiver_sig_required = 9;
+    /**
+     * The number of NFTs owned by the account.
+     */
+    int64 number_owned_nfts = 10;
+    /**
+     * The maximum number of tokens that can be auto-associated with the account.
+     */
+    int32 max_auto_associations = 11;
+    /**
+     * The number of used auto-association slots.
+     */
+    int32 used_auto_associations = 12;
+    /**
+     * The number of tokens associated with the account.
+     */
+    int32 number_associations = 13;
+    /**
+     * The number of tokens with a positive balance associated with the account.
+     */
+    int32 number_positive_balances = 14;
+    /**
+     * The nonce of the account, used for Ethereum interoperability.
+     */
+    int64 ethereum_nonce = 15;
+    /**
+     * The amount of hbars staked to the account.
+     */
+    int64 staked_to_me = 16;
+    /**
+     * If this account stakes to another account, its value will be -1. It will
+     * be set to the time when the account starts staking to a node.
+     */
+    int64 stake_period_start = 17;
+    /**
+     * The node number or the account number this account is staked to.
+     * It is negative if staking to a node and positive if staking to an account and 0 if not staking to anyone.
+     * When staking to a node, it is stored as -node-1 to differentiate node 0.
+     */
+    int64 staked_number = 18;
+    /**
+     * A boolean marking if the account declines rewards.
+     */
+    bool decline_reward = 19;
+    /**
+     * The amount of hbars staked to the account at the start of the last rewarded period.
+     */
+    int64 stake_at_start_of_last_rewarded_period = 20;
+    /**
+     * (Optional) The number of an auto-renew account, in the same shard and realm as the account, that
+     * has signed a transaction allowing the network to use its balance to automatically extend the account's
+     * expiration time when it passes.
+     */
+    int64 auto_renew_account_number = 21;
+    /**
+     * The number of seconds the network should automatically extend the account's expiration by, if the
+     * account has a valid auto-renew account, and is not deleted upon expiration.
+     */
+    int64 auto_renew_secs = 22;
+    /**
+     * If this account is a smart-contract, number of key-value pairs stored on the contract.
+     */
+    int32 contract_kv_pairs_number = 23;
+    /**
+     * (Optional) List of crypto allowances approved by the account.
+     */
+    repeated CryptoAllowance crypto_allowances = 24;
+    /**
+     * (Optional) List of non-fungible token allowances approved by the account.
+     */
+    repeated NftAllowance nft_allowances = 25;
+
+    /**
+     * (Optional) List of fungible token allowances approved by the account.
+     */
+    repeated TokenAllowance token_allowances = 26;
```

Comment:
I'm not sure these are right. It seems like we should be using our "map of linked list nodes" concept here instead. It isn't a blocker for this PR, but we need to discuss this.